### ISSUE TITLE
problems 2D classification

### ISSUE THREAD

**Issue by lauravanbezouwen** [3dem/relion#198]
We are trying to run a 2D classification. The first 9 iterations went fine. After that the following errors occured:

WARNING: Exception (filteredSize == 0) handled by switching to fail-safe mode.
Too many fail-safe attempts in one iteration
File: /src/gpu_utils/cuda_ml_optimiser.cu line: 3054

These we think we managed to solve by setting up the GPU different and using a smaller pool from 25 to 1. As we saw in a previous issue. We were hoping that was the problem.

But after that the iteration could never finish. Sometimes it just stops without giving an error back. Otherwise we had the following error: (based on these settings: `which relion_refine_mpi` --continue Class2D/job021/run_it009_optimiser.star --o Class2D/job021/run_ct9 --dont_combine_weights_via_disc --preread_images  --pool 1 --iter 25 --tau2_fudge 2 --particle_diameter 200 --oversampling 1 --psi_step 10 --offset_range 5 --offset_step 2 --j 2 --gpu "0,1"  and `which relion_refine_mpi` --continue Class2D/job021/run_it009_optimiser.star --o Class2D/job021/run_ct9 --dont_combine_weights_via_disc --scratch_dir /scratch/gideon --pool 1 --iter 25 --tau2_fudge 2 --particle_diameter 200 --oversampling 1 --psi_step 10 --offset_range 5 --offset_step 2 --j 2 --gpu "0,1" )

[cryo-em1:11593] *** Process received signal ***
[cryo-em1:11593] Signal: Segmentation fault (11)
[cryo-em1:11593] Signal code: Address not mapped (1)
[cryo-em1:11593] Failing at address: 0x147b8
[cryo-em1:11593] [ 0] /usr/lib64/libpthread.so.0(+0xf370)[0x7ff4570c3370]
[cryo-em1:11593] [ 1] /usr/lib64/libstdc++.so.6(_ZNSsC1ERKSs+0x1b)[0x7ff4578a6f4b]
[cryo-em1:11593] [ 2] /opt/RELION2/lib/librelion_lib.so(_ZN11RelionErrorC2ERKS_+0x1f)[0x7ff4602f70af]
[cryo-em1:11593] [ 3] /opt/RELION2/lib/librelion_lib.so(_ZN11MlOptimiser24expectationSomeParticlesEll+0x12c0)[0x7ff4602e2ba0]
[cryo-em1:11593] [ 4] /opt/RELION2/lib/librelion_lib.so(_ZN14MlOptimiserMpi11expectationEv+0x157a)[0x7ff460305e1a]
[cryo-em1:11593] [ 5] /opt/RELION2/lib/librelion_lib.so(_ZN14MlOptimiserMpi7iterateEv+0x5d)[0x7ff46030e29d]
[cryo-em1:11593] [ 6] /opt/RELION2/bin/relion_refine_mpi(main+0x69)[0x40a1b9]
[cryo-em1:11593] [ 7] /usr/lib64/libc.so.6(__libc_start_main+0xf5)[0x7ff456d14b35]
[cryo-em1:11593] [ 8] /opt/RELION2/bin/relion_refine_mpi[0x40a32f]
[cryo-em1:11593] *** End of error message ***

From time to time the following error would occur:

Array_by_array: different shapes (*)
File: /root/Downloads/RELION2/relion/src/multidim_array.h line: 2899

The problem is that we have no idea what happened. And we can not reproduce this error, at the moment that we wanted to write it down.

In august we have run successfully a 2D classification with all the default settings within relion2. In the mean time it has been re-installed, re-build for a new account to the most recent version. 

Can you help us out how get a good 2D classification?

Kind regards,

Laura

**Comment by bforsbe** [comment]
The array-by-array is something which is usually caused by a corrupt mrc header in our experience. It seems to surface every once in a while but without consistently reported cause. The only way to fix it is to re-generate the mrc-file causing the error, typically it's an input reference causing the error at the very beginning of a run.

As for the filtered-size=0, I'm a bit stumped, but trying to work it out. I've heard tales of --pool being the culprit but I've never seen it work to that effect and there is no logical reason why it would, given its function and implementation. I rather suspect some numerical instability associated with the precision, probably pertaining to the references. You could try to run with a different --seed number, just to nudge the algorithm ever so slightly. If you want to try it through --continue, you have to change the seed set in the optimiser.star file you continue from. 

In short: try the above, I'm on the case ;)

**Comment by lauravanbezouwen** [comment]
Thank you for your ideas. We first decided to restart the job but then it would crash again after a different amount of iterations. We tried to change the seed number but it didn't work. In the end based on  your comment about a corrupt mrc header, we decided to re-extract all particles. Then the job run without any problems at all.

Best wishes,
Laura

**Comment by bforsbe** [comment]
Regarding the extraction, was it using coordinates from relions manual/auto-picking or a different program? Or was it a re-extraction?

**Comment by lauravanbezouwen** [comment]
We just re-extracted the particles based on the coordinates. We did auto-picking with gautomatch and then imported the coordinates into the program.

**Comment by bforsbe** [comment]
Ok, good to know. I'll keep an eye open, it's good to have it documented anyhow.
